### PR TITLE
[Precision] Support lm_head layer running in float32

### PR DESCRIPTION
**Pull request by ckl117**
When service startup with `lm_head_fp32` , lm_head layer will run in float32.

### CODE REVIEW

File: fastdeploy/model_executor/layers/lm_head.py [hunk 1]
```diff
@@ -144,7 +154,7 @@ def forward(self, input: paddle.Tensor) -> paddle.Tensor:
         Returns:
             Tensor: The output tensor after processing through the layer.
         """
-        logits = input
+        logits = input.cast(self.dtype)
```
Comment (yuanlehome) [inline fastdeploy/model_executor/layers/lm_head.py]: 先判断一下如果不等于，再cast

File: fastdeploy/engine/args_utils.py [hunk 2]
```diff
@@ -560,6 +565,12 @@ def add_cli_args(parser: FlexibleArgumentParser) -> FlexibleArgumentParser:
             default=EngineArgs.early_stop_config,
             help="the config for early stop.",
         )
+        model_group.add_argument(
```
Comment (lizexu123) [inline fastdeploy/engine/args_utils.py:568]: 这个接口以后一直存在吗，如果后面被删了，要不要改成环境变量

File: fastdeploy/model_executor/layers/lm_head.py [hunk 3]
```diff
@@ -141,6 +148,9 @@ def forward(self, input: paddle.Tensor) -> paddle.Tensor:
         Returns:
             Tensor: The output tensor after processing through the layer.
         """
+        if input.dtype != self.linear.weight.dtype:
+            input = input.cast(self.linear.weight.dtype)
         logits = input
         logits = self.linear(logits)
+        print(f"self.linear.weight = {self.linear.weight}")
```
Comment (yuanlehome) [inline fastdeploy/model_executor/layers/lm_head.py]: delete

File: fastdeploy/model_executor/layers/lm_head.py [hunk 4]
```diff
@@ -39,6 +43,7 @@ def __init__(
         embedding_dim: int,
         prefix: str = "",
         with_bias: bool = False,
+        weight_dtype: str = paddle.get_default_dtype(),
```
Comment (yuanlehome) [inline fastdeploy/model_executor/layers/lm_head.py]: 名字就叫dtype就行，rms_norm我也加了个dtype参数，保持一致吧

File: fastdeploy/model_executor/models/deepseek_v3.py [hunk 5]
```diff
@@ -598,6 +598,7 @@ def __init__(self, fd_config: FDConfig):
             embedding_dim=fd_config.model_config.hidden_size,
             num_embeddings=fd_config.model_config.vocab_size,
             prefix="lm_head",
+            weight_dtype="float32" if self.fd_config.model_config.lm_head_fp32 else paddle.get_default_dtype(),
```
Comment (yuanlehome) [inline fastdeploy/model_executor/models/deepseek_v3.py]: 不要在这里设置传参呢，ParallelLMHead类里面是可以拿到 fd_config 的

File: fastdeploy/model_executor/models/deepseek_v3.py [hunk 6]
```diff
@@ -694,7 +695,8 @@ def load_weights(self, weights_iterator) -> None:
     def compute_logits(self, hidden_states: paddle.Tensor):
         """ """
         logits = self.lm_head(hidden_states)
-        logits = paddle.cast(logits, paddle.float32)
+        if not self.fd_config.model_config.lm_head_fp32:
+            logits = paddle.cast(logits, paddle.float32)
```
Comment (yuanlehome) [inline fastdeploy/model_executor/models/deepseek_v3.py]: 用 logits = logits.astype("float32")，如果一致内部不做cast，不一致自动会cast

**Review by yuanlehome — DISMISSED** [review]

**Review by Jiang-Jia-Jun — APPROVED** [review]
